pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,705 @@
+                attr['_href'],
+                cls.responses['distribute']
+            ) for attr in cls.attrs_iter
+        ))
+        cls.bodies['publish'] = tuple((
+            publish_repository(
+                cls.cfg,
+                cls.attrs_iter[i]['_href'],
+                cls.bodies['distribute'][i]['id'],
+                cls.responses['publish'],
+            ) for i in range(2)  # Both tuples are always of size 2
+        ))
+        for publish_report in cls.bodies['publish']:
+            for call_report in publish_report:
+                poll_spawned_tasks(cls.cfg, call_report)
+
+        # Form query as appears in puppet versions <=3.3
+        query_href_old = '/api/v1/releases.json?module={}/{}'.format(
+            _PUPPET_QUERY_PACKAGE['name'],
+            _PUPPET_QUERY_PACKAGE['author']
+        )
+        # Form query as appears in puppet 3.3 < version < 3.6
+        # This is list joined by repository name, ie. 'repoid'.join()
+        query_href_middle = [
+            '/pulp_puppet/forge/repository/',
+            '/api/v1/releases.json?module={}/{}'.format(
+                _PUPPET_QUERY_PACKAGE['name'],
+                _PUPPET_QUERY_PACKAGE['author']
+            )
+        ]
+        # Form query as appears in new puppet version (>=3.6).
+        query_href_new = '/v3/releases?module={}/{}'.format(
+            _PUPPET_QUERY_PACKAGE['name'],
+            _PUPPET_QUERY_PACKAGE['author']
+        )
+
+        # Query server using url as in puppet <= 3.3
+        cls.bodies['query units'] = tuple()
+        cls.bodies['query units'] += tuple(
+            _query_repo_old(
+                cls.cfg,
+                query_href_old,
+                body['repo_id'],
+                cls.responses['query units']
+            ) for body in cls.bodies['distribute']
+        )
+
+        # Query server using url as in puppet 3.3 > pup. version < 3.6
+        cls.bodies['query units'] += tuple(
+            _query_repo_middle(
+                cls.cfg,
+                body['repo_id'].join(query_href_middle),
+                cls.responses['query units']
+            ) for body in cls.bodies['distribute']
+        )
+
+        # Query server using new url format (puppet >= 3.6)
+        if cls.cfg.version > Version('2.6'):
+            cls.bodies['query units'] += tuple(
+                _query_repo_new(
+                    cls.cfg,
+                    query_href_new,
+                    body['repo_id'],
+                    cls.responses['query units']
+                ) for body in cls.bodies['distribute']
+            )
+
+        module_uris = []  # Save modules' uri on server
+        for query_body in cls.bodies['query units']:
+            try:
+                module_uris.append(
+                    query_body['{}/{}'.format(
+                        _PUPPET_QUERY_PACKAGE['name'],
+                        _PUPPET_QUERY_PACKAGE['author'])][0]['file']
+                )
+            except KeyError:
```

Comment:
Here I'm not sure what is best. The reason for try except block is: New forge API returns dictionary in difrerent format, so when trying to get module's uri on server, we got `KeyError` and thus second dictionary format is tried. I could avoid this by adding another key to `cls.bodies` or access them by index but I found first option (try except) least painful.